src/lib.rs

```diff
@@ -382,77 +432,83 @@ macro_rules! bitflags {
             }
         }
 
+        const _: () = {
```

Comment:
What's the reason for putting this in a `const _: () = { ... }`?

Reply:
It turns out `const _: () = { .. }` is a great container for code generated by macros because:

1. It's supported in any context where other items can be generated. You can write them in modules, functions, and the middle of other expressions.
2. It gives us a new scope to define items in. Any types we define in the body of this `const` [don't conflict with any outside of it](https://gist.github.com/027b2678e6cd7c6f3ceadb23d785f38e). Any types we define in the `const` are also unreachable from outside of it. That's why we've got a trait with an associated type that lets us peek inside. We can even wrap the contents of the `const` body in a `mod` to hide fields if we want.